app/services/product_tag_rules_filterer.rb

```diff
@@ -0,0 +1,107 @@
+class ProductTagRulesFilterer
+  def initialize(distributor, customer, variants)
```

Comment:
I think it's better to call this variants_relation to make it explicit this is a relation